wafflehaus/neutron/ip_policy/create_default.py

```diff
@@ -82,28 +86,68 @@ def _modify_allocation_pools(self, subnet):
         return alloc_pools
 
     def _filter_policy(self, req):
+        context_dict = req.environ.get('neutron.context')
+        if context_dict:
+            self.log.info('_check_basics, Neutron Context ' +
+                          'request id %s, project id %s, '
+                          'tenant name %s , is admin %s ,user id %s '
+                          % (str(context_dict.request_id),
+                             str(context_dict.project_id),
+                             str(context_dict.tenant_name),
+                             str(context_dict.is_admin),
+                             str(context_dict.user_id)))
+        tenant_id = req.headers.get('X_TENANT_ID')
+        user_id = req.headers.get('X_USER_ID')
         body = req.body
+        self.log.info('_filter_policy - '
+                      'Filter policy request body -> '
+                      '%s tenant_id %s and '
+                      'user_id %s ' % (str(body), tenant_id, user_id))
```

Comment:
Here and in many other places - there is an extra space at the end of line '%s '